File: Data Structure Array Programs in C, JS and Ruby/insertionAtIndex.js
// Program to insert an element at a specific index in an array

function display(arr, size) {
    // Array Traversal
  // Loop through array elements and print them
    for (let i = 0; i < size; i++) {
        console.log(`The value at index ${i} is ${arr[i]}`);
    }
}

function indexInsertion(arr, size, element, capacity, index) {
    // Check if the array is full
    if (size >= capacity) {
        return -1;
    }
    // Shift the elements to the right
    for (let i=size-1; i>=index; i--) {
        arr[i+1] = arr[i];
    }
    // Insert the element at the specified index
    arr[index] = element;
    console.log("Element inserted successfully");
    return 1;
}

function main() {
    let arr = [7, 8, 12, 27, 88];
    let size = 5;
    const element = 45, index = 3;
    display(arr, size);
    // Inserting element at index
    indexInsertion(arr, size, element, 100, index);
    size += 1;
    display(arr, size);
}

main();

File: Data Structure Array Programs in C, JS and Ruby/insertionAtIndex.rb
# Program to insert an element at a specific index in an array

def display(arr, size)
# Array traversal
# Loop through the array elements and print them
arr.first(size).each_with_index do |val, index|
    puts "The element at index #{index} is #{val}"
end
end

def insertAtIndex(arr, size, element, capacity, index)
 # Check if the array is full
    return -1 if size >= capacity

    #Shift the elements to the right
    size.downto(index+1) do |i|
        arr[i] = arr[i-1]
    end

    # Insert the element at the specified index
    arr[index] = element
    puts "Element inserted successfully"
    return 1

end

def main
# Array declaration
arr = [7, 8, 12, 27, 88]
size = arr.length
capacity = 100
index = 3
element = 45

# Display the original array
puts "Original Array"
display(arr, size)

# Inserting element at index 3
puts "Inserting element #{element} at index #{index}"
insertAtIndex(arr, size, element, capacity, index)
size += 1;

# Display the array after insertion
puts "Array after insertion"
display(arr, size)
end

main()

